### PR TITLE
Small updates

### PR DESCRIPTION
**Pull request by kevmoo**
Add cron
Align the flows between test and analyze

### CODE REVIEW

File: workflow-templates/test-package.yaml [hunk 1]
```diff
@@ -39,7 +42,6 @@ jobs:
     needs: analyze
     runs-on: ${{ matrix.os }}
     strategy:
-      fail-fast: false
```
Comment (athomas) [inline workflow-templates/test-package.yaml:42]: Removing this means that as soon as any of the jobs in the matrix fails, all other jobs will be cancelled immediately. That's different from what I've seen on travis.

File: workflow-templates/test-package.yaml [hunk 2]
```diff
@@ -28,6 +30,7 @@ jobs:
         run: dart pub get
       - name: Check formatting
         run: dart format --output=none --set-exit-if-changed .
+        if: always() && steps.install.outcome == 'success'
```
Comment (athomas) [inline workflow-templates/test-package.yaml:33]: That doesn't actually change anything, if any of the steps preceding this one fail, it makes no sense to run this either. But Ok for me if this is just for symmetry.

File: workflow-templates/test-package.yaml [hunk 3]
```diff
@@ -49,9 +51,12 @@ jobs:
       - uses: dart-lang/setup-dart@v0.1
         with:
           channel: ${{ matrix.sdk }}
-      - name: Install dependencies
+      - id: install
+        name: Install dependencies
         run: dart pub get
       - name: Run VM tests
         run: dart test --platform vm
+        if: always() && steps.install.outcome == 'success'
```
Comment (athomas) [inline workflow-templates/test-package.yaml:59]: Same here: That doesn't actually change anything, if any of the steps preceding this one fail, it makes no sense to run this either. But Ok for me if this is just for symmetry.

File: workflow-templates/test-package.yaml (hunk 1 above)
Comment (kevmoo) [reply workflow-templates/test-package.yaml:42]: Oh! So this makes sure this runs if the previous job fails. Got it.

Again, just for symmetry I'd add it above then

Comment (kevmoo) [reply workflow-templates/test-package.yaml]: fixed

**Review by athomas — APPROVED** [review]

**Comment by mit-mit** [comment]
@kevmoo I don't understand the reason for adding the `if:` statements; I personally find them confusing. But I could be missing something. Can you elaborate what you like about them?

**Comment by athomas** [comment]
> @kevmoo I don't understand the reason for adding the `if:` statements; I personally find them confusing. But I could be missing something. Can you elaborate what you like about them?

The `if:` statements are there to run all the steps even if one of them fails. E.g. if formatting fails, still run the analysis (so you don't have to fix the formatting to see the analysis errors). The point of adding them always even if there is no preceding step was that if you later add another step before, the step will still run even if that other step fails.

As to what it does:
`always()`: means this step should run no matter what happened before. If you don't put this in, the default is `success()` which means that if any prior step failed this step will not run.
`steps.install.outcome == 'success'`: The step with the id "install" needs to have the outcome "success" (`pub get` worked).

So `if: always() && steps.install.outcome == 'success'` means run the step always if `pub get` succeeded.

**Comment by kevmoo** [comment]
What @athomas said

**Comment by mit-mit** [comment]
Thanks, I (think) i get it!

**Pull request merged**